crates/ev-precompiles/src/mint.rs

```diff
@@ -0,0 +1,343 @@
+// Mint precompile
+
+use alloy::{sol, sol_types::SolInterface};
+use alloy_evm::{
+    precompiles::{Precompile, PrecompileInput},
+    revm::precompile::{PrecompileError, PrecompileId, PrecompileResult},
+    EvmInternals, EvmInternalsError,
+};
+use alloy_primitives::{address, Address, Bytes, U256};
+use revm::precompile::PrecompileOutput;
+use std::sync::OnceLock;
+
+sol! {
+    interface INativeToken {
+        function mint(address to, uint256 amount);
+        function burn(address from, uint256 amount);
+    }
+}
+
+pub const MINT_PRECOMPILE_ADDR: Address = address!("0x000000000000000000000000000000000000F100");
+
+/// A custom precompile that mints the native token
+#[derive(Clone, Debug, Default)]
+pub struct MintPrecompile {
+    admin: Address,
```

Comment:
Is the intension that this admin could be configured as the bridging token router contract?
The address of that contract would need to be pre-computed and deterministic.

https://getfoundry.sh/guides/deterministic-deployments-using-create2/